src/nbl/asset/utils/CGeometryCreator.cpp

```diff
@@ -40,6 +40,8 @@ CGeometryCreator::return_type CGeometryCreator::createCubeMesh(const core::vecto
 	{
 		retval.indexCount = 36u;
 		auto indices = core::make_smart_refctd_ptr<asset::ICPUBuffer>(sizeof(uint16_t)*retval.indexCount);
+		const auto newUsageFlags = indices->getUsageFlags() | asset::IBuffer::EUF_INDEX_BUFFER_BIT;
+		indices->setUsageFlags(newUsageFlags);
```

Comment:
@Erfan-Ahmadi assigned to you now

Reply:
@devshgraphicsprogramming It's very simple, because in the GeometryCreator example where @achalpandeyy was doing tests, `IGPUMeshBuffer` is created directly via it's constructor and not via `ICPUMeshBuffer`.

https://github.com/Devsh-Graphics-Programming/Nabla/blob/4549c258dfa4be38493271776953ad44c6757de4/include/nbl/asset/ICPUMeshBuffer.h#L189

https://github.com/Devsh-Graphics-Programming/Nabla/blob/4549c258dfa4be38493271776953ad44c6757de4/examples_tests/35.GeometryCreator/main.cpp#L442

I can remove this `addUsageFlags` from `CGeometryCreator.cpp` and handle it in the example, but I'm starting to think It may be a good idea to add the usages here even though they may be redundant in `CGeometryCreator.cpp` since we know these buffers are going to be used as Vertex/Index buffers.

Reply:
ah yeah, I forgot that one can skip turning `IGeometryCreator::return_type` into an `ICPUMeshBuffer` and the index buffer will never get the flags!